sunkit_magex/pfss/utils.py

```diff
@@ -1,111 +1,16 @@
-import os
 
 import numpy as np
 
 import astropy.constants as const
 import astropy.coordinates as coord
-import astropy.io
-import astropy.time
 from astropy import units as u
 from astropy.wcs import WCS
 
 import sunpy.coordinates
 import sunpy.map
 import sunpy.time
 
-__all__ = ['fix_hmi_meta', 'load_adapt', 'carr_cea_wcs_header', 'is_cea_map', 'is_car_map', 'is_full_sun_synoptic_map', 'car_to_cea', 'roll_map']
-
-def _observer_coord_meta(observer_coord):
-    """
-    Convert an observer coordinate into FITS metadata.
-    """
-    new_obs_frame = sunpy.coordinates.HeliographicStonyhurst(
-        obstime=observer_coord.obstime)
-    observer_coord = observer_coord.transform_to(new_obs_frame)
-
-    new_meta = {'hglt_obs': observer_coord.lat.to_value(u.deg)}
-    new_meta['hgln_obs'] = observer_coord.lon.to_value(u.deg)
-    new_meta['dsun_obs'] = observer_coord.radius.to_value(u.m)
-    return new_meta
-
-
-def _earth_obs_coord_meta(obstime):
-    """
-    Return metadata for an Earth observer coordinate.
-    """
-    return _observer_coord_meta(sunpy.coordinates.get_earth(obstime))
-
-
-def fix_hmi_meta(hmi_map):
-    """
-    Fix non-compliant FITS metadata in HMI maps.
-
-    This function:
-        - Corrects CUNIT2 from 'Sine Latitude' to 'deg'
-        - Corrects CDELT1 and CDELT2 for a CEA projection
-        - Populates the DATE-OBS keyword from the T_OBS keyword
-        - Sets the observer coordinate to the Earth
-
-    Notes
-    -----
-    If you have sunpy > 2.1 installed, this function is not needed as sunpy
-    will automatically make these fixes.
-    """
-    if not isinstance(hmi_map, sunpy.map.sources.HMIMap):
-        raise ValueError('Input must be of type HMIMap. '
-                         'n.b. if you have sunpy 2.1 installed, '
-                         'this function is redundant '
-                         'as sunpy 2.1 automatically fixes HMI metadata.')
-
-    if hmi_map.meta['cunit1'] == 'Degree':
-        hmi_map.meta['cunit1'] = 'deg'
-
-    if hmi_map.meta['cunit2'] == 'Sine Latitude':
-        hmi_map.meta['cunit2'] = 'deg'
-
-        # Since, this map uses the cylindrical equal-area (CEA) projection,
-        # the spacing should be modified to 180/pi times the original value
-        # Reference: Section 5.5, Thompson 2006
-        hmi_map.meta['cdelt2'] = 180 / np.pi * hmi_map.meta['cdelt2']
-        hmi_map.meta['cdelt1'] = np.abs(hmi_map.meta['cdelt1'])
-
-    if 'date-obs' not in hmi_map.meta and 't_obs' in hmi_map.meta:
-        hmi_map.meta['date-obs'] = sunpy.time.parse_time(
-            hmi_map.meta['t_obs']).isot
-
-    # Fix observer coordinate
-    if 'hglt_obs' not in hmi_map.meta:
-
-        hmi_map.meta.update(_earth_obs_coord_meta(hmi_map.meta['date-obs']))
-
-
-def load_adapt(adapt_path):
```

Comment:
We dont have the same loader in sunpy but we can now load them directly. But only the first one.

Reply:
I don't know enough about the data to know if the other ones are useful, but if there isn't a drop in replacement then maybe we shouldn't yank it?

Also deprecations please :angel:

Reply:
Added.